extension/src/popup/components/sendPayment/SendConfirm/TransactionDetails/index.tsx

```diff
@@ -357,7 +357,9 @@ export const TransactionDetails = ({ goBack }: { goBack: () => void }) => {
           url,
           networkDetails,
         );
+        return;
       }
+      setLoading(false);
```

Comment:
do we need to worry about this loading state for classic tx's, as well?

Reply:
no we don't because we seem to be calling that in every case here. I guess maybe I should about the intention of the condition to not `scanSorobanTx` if `submission.submitStatus === ActionStatus.IDLE && transactionSimulation.preparedTransaction`? I think that's actually correct since we don't really get anything from re-scanning it in tx details after the tx has been done. Maybe the miss here is that we should also be doing this in `scanClassicTx`?

Reply:
oh wait I wrote that! lol Yeah curious if you think that's right? I think we probably also want to skip the scan for classic in the same condition.

Reply:
Okay took me a second to understand what was happening, but I get it now:

This conditional is just to make sure we've finished simming the tx, which is only necessary for Soroban tx's. So it makes sense that we only have the conditional on scanSorobanTx

Reply:
I changed this in fd2e8dcc617d3c27b72ff2d0ca2de6084d75044f
I think it makes sense to not scan either tx type in tx details if we are rendering it after the tx has been sent.

Reply:
ah maybe I misunderstand this conditional. I think really what we want here is to never scan the tx is we are rendering the details at confirmation so I will just add a prop that we can pass in explicitly at confirmation to make this less brittle.

Reply:
But if we go back to tx details from the Tx Success screen, we'll re-scan a classic tx, but we won't re-scan a soroban tx. And that's kind of a weird discrepancy. We probably shouldn't re-scan in either case

Reply:
ok cool I think we're aligned here, 62cd1a6cd66f2a512d4b2850cf08932d01e6eaee is the version that I think handles this best.

Reply:
Nice - this lgtm!